### ISSUE TITLE
What is the minimum GPU requirement in gigabytes for TRL intensive training?

### ISSUE THREAD

**Issue by lonngxiang** [huggingface/trl#2813]

**Comment by Superskyyy** [comment]
Your issue isn't complete. What do you mean by intensive? The GPU requirement highly depends on dataset characteristics and algorithm to use.

**Comment by lonngxiang** [comment]
For example, what is the minimum amount of GPU resources required for the official reinforcement learning examples?
```
from datasets import load_dataset
from trl import GRPOConfig, GRPOTrainer

dataset = load_dataset("trl-lib/tldr", split="train")

# Dummy reward function: rewards completions that are close to 20 characters
def reward_len(completions, **kwargs):
    return [-abs(20 - len(completion)) for completion in completions]

training_args = GRPOConfig(output_dir="Qwen2-0.5B-GRPO", logging_steps=10)
trainer = GRPOTrainer(
    model="Qwen/Qwen2-0.5B-Instruct",
    reward_funcs=reward_len,
    args=training_args,
    train_dataset=dataset,
)
trainer.train()
```

**Comment by lonngxiang** [comment]
Using the free Colab T4, it's not working. Could you please provide some test code?

**Comment by qgallouedec** [comment]
Hey @lonngxiang, sorry for the slow reply, catching up on the backlog.

GPU memory requirements depend heavily on the model size, sequence length, batch size vs gradient accumulation, whether you use PEFT/LoRA, etc., so there's no single "minimum" number we can put in the docs. As @Superskyyy noted, the example you posted should fit in ~16 GB with batch size 1, short `max_completion_length` but a free Colab T4 is genuinely tight for GRPO. Closing here.
